overlord/restart/restart_parameters.go

```diff
@@ -0,0 +1,82 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package restart
+
+import (
+	"github.com/snapcore/snapd/boot"
+	"github.com/snapcore/snapd/bootloader"
+	"github.com/snapcore/snapd/snap"
+)
+
+type RestartParameters struct {
+	SnapName        string      `json:"snap-name,omitempty"`
+	RestartType     RestartType `json:"restart-type,omitempty"`
+	RebootArguments string      `json:"reboot-args,omitempty"`
```

Comment:
Would it make sense to store instead `rebootInfo.RebootBootloader`, instead of caching the reboot arguments? You could compose with the `bootloader.RebootBootloader` type so you do not need to implement the methods.

Reply:
The challenge is that `rebootInfo.RebootBootloader` is not marshallable/unmarshallable to json